confidential/confidential.go

```diff
@@ -107,16 +113,21 @@ type RangeProofArgs struct {
 	ValueBlindFactor    [32]byte
 	ValueCommit         []byte
 	ScriptPubkey        []byte
-	MinValue            uint64
 	Exp                 int
 	MinBits             int
 }
 
 func (a RangeProofArgs) minValue() uint64 {
-	if a.MinValue <= 0 {
-		return 1
+	if isUnSpendable(a.ScriptPubkey) {
+		return 0
 	}
-	return a.MinValue
+
+	return 1
+}
```

Comment:
did you take this from elements core codebase?

Reply:
Yeah, check comment in PR decs.